lms/security.py

```diff
@@ -134,28 +172,24 @@ def _get_lti_user(request):
 
     :rtype: models.LTIUser
     """
-    bearer_token_schema = BearerTokenSchema(request)
-
-    schemas = [
-        LTI11AuthSchema(request).lti_user,
-        partial(bearer_token_schema.lti_user, location="headers"),
-        partial(bearer_token_schema.lti_user, location="querystring"),
-        partial(bearer_token_schema.lti_user, location="form"),
-        OAuthCallbackSchema(request).lti_user,
-    ]
-    # Avoid checking for the LTI1.3 JWT if not there
-    if "id_token" in request.params:
-        # Don't replace all auth methods in case somehow we have a bogus token
-        # but try this one first.
-        schemas.insert(0, LTI13AuthSchema(request).lti_user)
-
     lti_user = None
-    for schema in schemas:
-        try:
-            lti_user = schema()
-            break
-        except ValidationError:
-            continue
+    if from_identity and request.identity and request.identity.lti_user:
+        lti_user = request.identity.lti_user
+    else:
```

Comment:
Adding this `else` to still do the user_service call at the bottom in both cases.